src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -147,17 +149,25 @@ function setupScriptProcessor(recordingProperties, passSampleToVisualizers) {
  */
 function setupRecording(recordingProperties) {
   recordButton.addEventListener('click', (event) => {
-    isRecording = !isRecording;
+    isRecording = true;
+    changeButtonDisabled();
+  });
 
+  stopButton.addEventListener('click', (event) => {
     // When recording is paused, process clip.
-    if (!isRecording) {
-      const finalRecordBuffer = createFinalRecordBuffer(recordingProperties);
-      prepareClip(finalRecordBuffer);
-    }
-
-    recordText.textContent = isRecording ? 'Stop' : 'Start';
-    downloadButton.disabled = isRecording ? true: false;
+    isRecording = false;
+    const finalRecordBuffer = createFinalRecordBuffer(recordingProperties);
+    prepareClip(finalRecordBuffer);
+    changeButtonDisabled();
   });
+
+
+}
+
+function changeButtonDisabled() {
```

Comment:
The naming scheme in Chromium would be:
- `changeButtonStatusIfNeeded()`
- `maybeChangeButtonStatus()`

This suggests: this function may or may not change the status of buttons based on some factors.